### ISSUE TITLE
.NET 6 AmbiguousMatchException Mixed Technology App

### ISSUE THREAD

**Issue by fceci90** [dotnet/aspnetcore#45493]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Pretty much the same exact bug as issue [38566 ](https://github.com/dotnet/aspnetcore/issues/38566) which was marked as closed without being resolved.

When upgrading an app from .NET 3.1 to .NET 6 or .NET 5 to .NET 6 which includes mixed technologies (with Blazor being one of them), routing fails to work correctly on Blazor pages. By mixed technologies I mean Razor Pages, MVC, and Blazor. In my particular case I am using MVC, Razor Pages, and Blazor Server. 

Downgrading from Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation version 6.0.11 to version 5.0.17 (latest version for 5) _"fixes"_ the issue, as discussed in the prior bug linked above. Version 6 has gone from 6.0.0 to 6.0.11 in the past year and the bug still exists. I am confused as to why it was closed.

If using a .NET 6 shouldn't I also be using version 6 of all my .NET dependencies?

### Expected Behavior

For routing to work properly as it did in .NET Core 3.1 and .NET 5 as there are no duplicate routes.

### Steps To Reproduce

Upgrade a .NET Core 3.1 to .NET 5 app included Razor pages, MVC, and Blazor Server to a .NET 6 app.

### Exceptions (if any)

Microsoft.AspNetCore.Routing.Matching.AmbiguousMatchException: The request matched multiple endpoints. Matches: 
/_Host
/_Host
   at Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.ReportAmbiguity(CandidateState[] candidateState)
   at Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.ProcessFinalCandidates(HttpContext httpContext, CandidateState[] candidateState)
   at Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.Select(HttpContext httpContext, CandidateState[] candidateState)
   at Microsoft.AspNetCore.Routing.Matching.DefaultEndpointSelector.SelectAsync(HttpContext httpContext, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)

### .NET Version

6

### Anything else?

Visual Studio Version 17.4.2

**Comment by davidfowl** [comment]
Typically when things are closed like that it's because of missing information. How do we reproduce the issue?

> Upgrade a .NET Core 3.1 to .NET 5 app included Razor pages, MVC, and Blazor Server to a .NET 6 app.

Can you clarify these repro steps for us? Can you reproduce this 100% of the time when doing this from scratch? It's unclear what *exact* steps you took during the upgrade to reproduce the issue. Its also possible that the upgrade guide you followed put the application in a state where this happened. 

Without repro steps, or an application that shows the problem that the team can run, this issue will end up the same as the other one.

**Comment by fceci90** [comment]
Hi David,

Thank you for your response. I was just able to resolve the issue. The hosting server only had .NET 6.0.1 installed on it (which was the version previously reported last year to have this issue). Updating the hosting bundle to 6.0.11 resolved this issue, so this issue must have been resolved at some point in the past year.

My MVC routing is still broken after the upgrade, and I'm not sure how, but I've just implemented attribute routing for now since new development will most likely be in Blazor.

Feel free to close this issue.

For any relevant information that you might be curious in:
Microsoft .NET Core 3.1 to .NET 6 [migration guide](https://learn.microsoft.com/en-us/aspnet/core/migration/31-to-60?view=aspnetcore-6.0&tabs=visual-studio)
How to Include Blazor in a .NET Core 3.1 MVC app [YouTube Video from DevExpress](https://www.youtube.com/watch?v=rfPQytn7tfk)
How to Include Blazor in a .NET Core 3.1 MVC app [source code on GitHub](https://github.com/kinetiq/Blazor.WithMVC/).

Thanks again and I appreciate your time.
